api/repositories/build_repository_test.go

```diff
@@ -595,12 +613,104 @@ var _ = Describe("BuildRepository", func() {
 				createRoleBinding(ctx, userName, spaceDeveloperRole.Name, namespace1.Name)
 			})
 			It("if user is a Space Developer it returns a list with one element", func() {
-				buildRecords, fetchError = buildRepo.ListBuilds(ctx, authInfo)
+				buildRecords, fetchError = buildRepo.ListBuilds(ctx, authInfo, listMessage)
 				Expect(fetchError).NotTo(HaveOccurred())
 				Expect(buildRecords).To(HaveLen(1))
 				Expect(buildRecords[0].GUID).To(Equal(build1GUID))
 			})
 		})
+		When("the app_guids filter is provided", func() {
+			BeforeEach(func() {
+				createRoleBinding(ctx, userName, spaceDeveloperRole.Name, namespace1.Name)
+				listMessage = repositories.ListBuildsMessage{AppGUIDs: []string{app1GUID}}
+			})
+			It("fetches all BuildRecords for that app", func() {
+				buildRecords, fetchError = buildRepo.ListBuilds(ctx, authInfo, listMessage)
+				Expect(fetchError).NotTo(HaveOccurred())
+				Expect(buildRecords).To(HaveLen(1))
```

Comment:
You could use 

```
Expect(buildRecords).To(ConsistOf(gstruct.MatchFields(gstruct.IgnoreExtras, gstruct.Fields{
	"AppGUID": Equal(app1GUID),
}))
```

This verifies that the slice has a single element that matches the `MatchFields` matcher. Same goes for the other filtering contexts